### ISSUE TITLE
Http StatusCode always 200

### ISSUE THREAD

**Issue by fingthinking** [dapr/components-contrib#1425]
I load ratelimit component to my dapr runtime. But it return http status code with 200 always whatever the server response is success or failure.

## Expected Behavior

return status code not 200 when the server response is failure.

## Actual Behavior

return http status code with 200 always whatever the server response is success or failure.

## Steps to Reproduce the Problem

different with stack

step 1: no ratelimit
![image](https://user-images.githubusercontent.com/5517388/148675861-b669b28f-1218-4f43-ba31-47b52629e111.png)

step 2: with ratelimit
![image](https://user-images.githubusercontent.com/5517388/148675887-b0395b5a-669c-4ecd-b3e1-0fd2e77b8e2b.png)

config yaml
```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: ratelimit
spec:
  version: v1
  type: middleware.http.ratelimit
  metadata:
    - name: maxRequestsPerSecond
      value: 100000
```
## Problem

I thank it's reason is in [nethttpadaptor](https://github.com/dapr/components-contrib/blob/master/middleware/http/nethttpadaptor/nethttpadaptor.go).

This code didn't process response status code.

**Comment by daixiang0** [comment]
@fingthinking Check the code, the statusCode is returned by "github.com/didip/tollbooth", if you do not reach limit, it would return no error.

**Comment by fingthinking** [comment]
> @fingthinking Check the code, the statusCode is returned by "github.com/didip/tollbooth", if you do not reach limit, it would return no error.

OK! Thank you for your answer. But even if I actively set the status code to 500, the status code is always 200, which is not what I expected. I think it is a bug. Do you have another point of view?

**Comment by jjcollinge** [comment]
I'll try to repro this today and push a fix if needed.

**Comment by jjcollinge** [comment]
This is indeed a bug, thanks for reporting, I've added a test and a fix for this in #1429

**Comment by fingthinking** [comment]
> This is indeed a bug in the adaptor, thanks for reporting, I think I've fix this in #1429

LGTM, thank u. I will test this PR.
